src/gloss/data/primitives.clj

```diff
@@ -61,6 +61,38 @@
   [x]
   (.longValue (bigint x)))
 
+(defn bytes-umedium
+  [x]
+  (let [[b0 b1 b2] x]
+    (bit-or
+     (bit-or
+      (bit-shift-left (bit-and b0 0xFF) 16)
+      (bit-shift-left (bit-and b1 0xFF) 8))
+     (bit-shift-left (bit-and b2 0xFF) 0))))
+
+(defn umedium-medium
+  [x]
+  (let [uval2 (bit-and x 0x800000)]
+    (if (not= uval2 0)
+      (bit-or x 0xff000000)
+      x)))
+
+(defn medium-bytes
+  [x]
+  (into-array Byte/TYPE
+   (map byte
+        [(bit-shift-right x 16)
+         (bit-shift-right x 8)
+         (bit-shift-right x 0)])))
+
+(defn getMedium [buffer]
+  (let [bytes (byte-array 3)]
+    (.get buffer bytes)
+    (bytes-umedium bytes)))
+
+(defn putMedium [buffer value]
+  (let [bytes (medium-bytes value)]
+    (.put buffer bytes)))
```

Comment:
Since these aren't actually methods on `ByteBuffer`, I'd prefer to keep them Lisp kebab-cased, like `get-medium` and `put-medium`.

Reply:
No problem. I'm also going to check if type hint is required here.